tests/value_ptr.cpp

```diff
@@ -0,0 +1,45 @@
+#include "catch/catch.hpp"
+#include "value_ptr.h"
+
+TEST_CASE( "value_ptr copy constructor", "[value_ptr]" )
+{
+    cata::value_ptr<int> a = cata::make_value<int>( 7 );
+    REQUIRE( !!a );
+    cata::value_ptr<int> b( a );
+    CHECK( !!a );
+    CHECK( !!b );
+    a.reset( nullptr );
+    CHECK( !a );
+    CHECK( !!b );
+}
+
+TEST_CASE( "value_ptr copy assignment", "[value_ptr]" )
+{
+    cata::value_ptr<int> a = cata::make_value<int>( 7 );
+    REQUIRE( !!a );
+    cata::value_ptr<int> b;
+    REQUIRE( !b );
+    b = a;
+    CHECK( !!a );
+    CHECK( !!b );
+    a.reset( nullptr );
+    CHECK( !a );
+    CHECK( !!b );
+}
+
+TEST_CASE( "value_ptr move constructor", "[value_ptr]" )
+{
+    cata::value_ptr<int> a = cata::make_value<int>( 7 );
+    REQUIRE( !!a );
+    cata::value_ptr<int> b( std::move( a ) );
+    CHECK( !a );
```

Comment:
You will probably need to add a suppression here to avoid a clang-tidy warning about use-after-move.

Reply:
Alright, I believe I've suppressed it correctly